src/PartielsPy/partiels.py

```
@@ -63,29 +73,79 @@ def __init__(self):
+    @property
+    def executable_version(self):
         """Return Partiels's executable version"""
-
         return self.__executable_version
 
-    def getCompatibilityVersion(self):
+    @property
+    def compatibility_version(self):
         """Return the PartielsPy's compatibility version"""
-
         return self.__compatibility_version
+
+    def get_default_template(self, name) -> str:
+        """Get a default template path
+
+        This method is used to get a default template path.
+        The default templates are stored in the PartielsPy package.
+        Default templates are:
+        - factory
+        - harmonic_partials_tracking
+        - waveform
+
+        Args:
+            name (str): the name of the template
+
+        Returns:
+            str: the path to the template
+        """
+        if name in (f.value for f in self.DefaultTemplateEnum):
+            return pkg_resources.resource_filename(
+                __name__, "templates/" + name + ".ptldoc"
+            )
+        else:
+            raise ValueError(f"template {name} is not a valid template name.")
+
+    def export(self, audiofile_path, template_path, output_path, ExportConfig) -> int:
+        """Export the audiofile with the template and export configuration
+
+        Args:
+            audiofile_path (str): the path to the audio file
+            template_path (str): the path to the template
+            output_path (str): the path to the output folder
+            Export_Config (Export_Config): the export configuration
+        """
+        cmd = [
+            self.__executable_path,
+            "--export",
+            "--input=" + audiofile_path,
+            "--template=" + template_path,
+            "--output=" + output_path,
+        ]
+        cmd += ExportConfig.getArgs()
+        print(cmd)
+        ret = subprocess.run(cmd, capture_output=True, text=True, check=True)
+        if ret.stdout:
+            print(ret.stdout)
+        return ret.returncode
```

Following the specifications, you should raise an exception if the return code isn't 0

---

The check flag of subprocess.run specifies that if the return value is not 0 an exception is raised.

---

In this case, if an execption is raised, the program will stop at the line 148 so the remaining part is useless

---

but i think it is nice to keep the print of the stdout if the export run correctly.